backtester/engine/grpcserver.go

```diff
@@ -0,0 +1,500 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	grpcauth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/shopspring/decimal"
+	"github.com/thrasher-corp/gocryptotrader/backtester/btrpc"
+	"github.com/thrasher-corp/gocryptotrader/backtester/common"
+	"github.com/thrasher-corp/gocryptotrader/backtester/config"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/database"
+	"github.com/thrasher-corp/gocryptotrader/database/drivers"
+	gctengine "github.com/thrasher-corp/gocryptotrader/engine"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	gctkline "github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/gctrpc/auth"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/utils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+)
+
+// GRPCServer struct
+type GRPCServer struct {
+	btrpc.BacktesterServer
+	*config.BacktesterConfig
+}
+
+// SetupRPCServer sets up the gRPC server
+func SetupRPCServer(cfg *config.BacktesterConfig) *GRPCServer {
+	return &GRPCServer{
+		BacktesterConfig: cfg,
+	}
+}
+
+// StartRPCServer starts a gRPC server with TLS auth
+func StartRPCServer(server *GRPCServer) error {
+	targetDir := utils.GetTLSDir(server.GRPC.TLSDir)
+	if err := gctengine.CheckCerts(targetDir); err != nil {
+		return err
+	}
+	log.Debugf(log.GRPCSys, "gRPC server support enabled. Starting gRPC server on https://%v.\n", server.GRPC.ListenAddress)
```

Comment:
Maybe add in backtester strings to logs, just for differentiation between the main RPC server?

Reply:
Could you clarify this one?

Reply:
So maybe do `log.Debugf(log.GRPCSys, "gRPC backtester server support enabled. Starting gRPC server on https://%v.\n", server.GRPC.ListenAddress)` but it probably doesn't matter 🤷 now thinking about it.